checkov/terraform/parser.py

```diff
@@ -512,15 +535,16 @@ def _load_modules(self, root_dir: str, module_loader_registry: ModuleLoaderRegis
                             #       has not already been added.
                             keys = list(module_definitions.keys())
                             for key in keys:
-                                if key.endswith("]"):
+                                if key.endswith("]") or file.endswith("]"):
                                     continue
+                                keys_referenced_as_modules.add(key)
                                 new_key = f"{key}[{file}#{module_index}]"
-                                module_definitions[new_key] = \
-                                    module_definitions[key]
+                                module_definitions[new_key] = module_definitions[key]
                                 del module_definitions[key]
                                 del self.out_definitions[key]
-
-                                resolved_loc_list.append(new_key)
+                                if new_key not in resolved_loc_list:
+                                    resolved_loc_list.append(new_key)
+                            resolved_loc_list.sort()        # For testing, need predictable ordering
```

Comment:
@robeden I've found the predictable ordering is important in a multitude of cases, +1 for that!